awswrangler/dynamodb/_read.py

```diff
@@ -146,3 +150,331 @@ def read_partiql_query(
+            # Check catched error to verify its message
+            kwarg = _get_invalid_kwarg(error_message)
+            if (error_code == "ValidationException") and kwarg:
+                reserved_keyword = error_message.split("keyword: ")[-1]
+                sanitized_keyword = f"#{reserved_keyword}"
+                kwargs[kwarg] = kwargs[kwarg].replace(reserved_keyword, sanitized_keyword)
+                kwargs["ExpressionAttributeNames"] = {
+                    **kwargs.get("ExpressionAttributeNames", {}),
+                    sanitized_keyword: reserved_keyword,
+                }
+                # SEE: recursive approach guarantees that each reserved keyword will be properly replaced,
+                # even if it will require as many calls as the reserved keywords involved (not so efficient...)
+                return wrapper(*args, **kwargs)
+            # Otherwise raise it
+            raise e
+
+    # SEE: https://github.com/python/mypy/issues/3157#issue-221120895
+    return cast(CustomCallable, wrapper)
+
+
+@_handle_reserved_keyword_error
+def _read_items(
+    table_name: str, boto3_session: Optional[boto3.Session] = None, **kwargs: Any
+) -> Sequence[Dict[str, Any]]:
+    """Read items from given DynamoDB table.
+
+    This function set the optimal reading strategy based on the received kwargs.
+
+    Parameters
+    ----------
+    table_name : str
+        DynamoDB table name.
+    boto3_session : boto3.Session, optional
+        Boto3 Session. Defaults to None (the default boto3 Session will be used).
+
+    Returns
+    -------
+    Sequence[Mapping[str, Any]]
+        Retrieved items.
+    """
+    # Get DynamoDB resource and Table instance
+    resource = _utils.resource(service_name="dynamodb", session=boto3_session)
+    table = get_table(table_name=table_name, boto3_session=boto3_session)
+
+    # Extract 'Keys' from provided kwargs: if needed, will be reinserted later on
+    keys = kwargs.pop("Keys", None)
+
+    # Conditionally define optimal reading strategy
+    use_get_item = (keys is not None) and (len(keys) == 1)
+    use_batch_get_item = (keys is not None) and (len(keys) > 1)
+    use_query = (keys is None) and ("KeyConditionExpression" in kwargs)
+    use_scan = (keys is None) and ("KeyConditionExpression" not in kwargs)
+
+    # Read items
+    if use_get_item:
+        kwargs["Key"] = keys[0]
+        items = [table.get_item(**kwargs).get("Item", {})]
+    elif use_batch_get_item:
+        kwargs["Keys"] = keys
+        response = resource.batch_get_item(RequestItems={table_name: kwargs})
+        items = response.get("Responses", {table_name: []}).get(table_name, [])
+        # SEE: handle possible unprocessed keys. As suggested in Boto3 docs,
+        # this approach should involve exponential backoff, but this should be
+        # already managed by AWS SDK itself, as stated
+        # [here](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Programming.Errors.html)
```

Comment:
I agree that it should be handled by the SDK. If not, we do have a [method](https://github.com/aws/aws-sdk-pandas/blob/main/awswrangler/_utils.py#L333) which is used for exponential backoff. It requires knowing the exception though